### PR TITLE
Split requirements.txt into 2 files

### PR DESCRIPTION
**Pull request by nhandler**
Split requirements.txt into requirements.txt and test-requirements.txt.
test-requirements.txt is only used by tox for testing purposes.
Modules in requirements.txt are needed to actually run/use the code.

### CODE REVIEW

**Comment by nhandler** [comment]
@Yelp/paasta this is needed because of testifycompat.

**Comment by mrtyler** [comment]
Seems fine but I don't know all the ramifications of this change. I'd consult with a yelp_clog and/or tox/setup.py/python-packaging-insanity expert.

**Comment by nhandler** [comment]
@ecanzonieri or @yyejun: Any thoughts on this?

File: tox.ini [hunk 1]
```diff
@@ -5,7 +5,9 @@ indexserver =
         default = https://pypi.yelpcorp.com/simple/
 
 [testenv]
-deps = -rrequirements.txt
+deps =
+	-rrequirements.txt
+	-rtest-requirements.txt
```
Comment (solarkennedy) [inline tox.ini]: This is totally fine and a common pattern. Usually the file here is called requirements-dev.txt
https://github.com/search?q=requirements-dev.txt&type=Code&utf8=%E2%9C%93

**Comment by nhandler** [comment]
@solarkennedy's comment is hidden with the outdated diff, but I have gone ahead and renamed test-requirements.txt -> requirements-dev.txt to match convention as he suggested. The various makefile targets and tox environments all appears to still work. I am going to merge this.

**Pull request merged**